event-adapter/cmd/provider/provider.go

```diff
@@ -17,44 +17,146 @@ limitations under the License.
 package provider
 
 import (
+	gce "cloud.google.com/go/compute/metadata"
 	"fmt"
 	"github.com/GoogleCloudPlatform/k8s-stackdriver/event-adapter/pkg/provider"
-	stackdriver "google.golang.org/api/monitoring/v3"
-	"k8s.io/client-go/pkg/api"
+	sd "google.golang.org/api/logging/v2"
+	api "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
+	"time"
 )
 
+//TODO(erocchi) add implementation to write to Stackdriver
 //TODO(erocchi) Use the Kubernetes AIP - compatible errors, see example definitions here: https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/pkg/provider/errors.go
 
-type sdService stackdriver.Service
-
 // StackdriverProvider implements EventsProvider
 type StackdriverProvider struct {
-	//TODO(erocchi) Define fields for this struct. It should contain what will be needed to communicate with Stackdriver.
+	restClient         rest.Interface
+	service            *sd.Service
```

Comment:
Maybe "stackdriverService" or "sdService"? Just "service" is too ambiguous.

Reply:
Done